src/js/components/AppContainer/AppContainer.js

```diff
@@ -0,0 +1,58 @@
+import React, { PureComponent } from 'react';
+import {
+  HashRouter as Router, Route, Switch, Redirect
+} from 'react-router-dom';
+import { isUserAuth } from '../../helpers';
+import Header from '../Header';
+import Footer from '../Footer';
+import Courses from '../Courses';
+import LoginPage from '../LoginPage';
+import ErrorPage from '../ErrorPage';
+
+class AppContainer extends PureComponent {
+    state = {
+      isAuth: false
+    };
+
+    componentDidMount() {
+      this.setState({ isAuth: isUserAuth() });
+    }
+
+    render() {
+      const authProps = {
+        isAuth: this.state.isAuth,
+        setAuth: isLogin => this.setState({ isAuth: isLogin })
+      };
+      return (
+        <Router>
+          <Header
+            {...authProps}
+          />
+          <Switch>
+            <Route
+              path="/"
+              exact
+              {...authProps}
+            >
+              <Redirect to="/courses" {...authProps} />
+            </Route>
+            <Route
+              path="/login"
+              {...authProps}
```

Comment:
why you put this props here if you already passed them through the component props?